federation/source-config-handlers/src/main/java/org/codice/ddf/admin/sources/opensearch/OpenSearchSourceUtils.java

```diff
@@ -54,89 +53,67 @@
 
     private static final String TOTAL_RESULTS_XPATH = "//os:totalResults|//opensearch:totalResults";
 
-    //Given a config, returns the correct URL format for the endpoint if one exists
-    public UrlAvailability confirmEndpointUrl(OpenSearchSourceConfiguration config) {
-        Optional<UrlAvailability> result = URL_FORMATS.stream()
-                .map(formatUrl -> String.format(formatUrl,
-                        config.sourceHostName(),
-                        config.sourcePort()))
-                .map(url -> getUrlAvailability(url, config.sourceUserName(), config.sourceUserPassword()))
-                .filter(avail -> avail.isAvailable() || avail.isCertError())
-                .findFirst();
-        return result.isPresent() ? result.get() : null;
-    }
+    /**
+     * Confirms whether or not an endpoint has Opensearch capabilities.
+     * SUCCESS TYPES - VERIFIED_CAPABILITIES,
+     * FAILURE TYPES - CANNOT_CONNECT, CERT_ERROR, UNKNOWN_ENDPOINT
+     * WARNING TYPES - UNTRUSTED_CA
+     * RETURN TYPES -  CONTENT_TYPE, CONTENT, STATUS_CODE
+     *
+     * @param url
+     * @return report
+     */
+    public static ProbeReport verifyOpenSearchCapabilities(String url, String username, String password) {
+        ProbeReport requestResults = RequestUtils.sendGetRequest(url + SIMPLE_QUERY_PARAMS, username, password);
+        if(requestResults.containsFailureMessages()) {
+            return requestResults;
+        }
+
+        int statusCode = requestResults.getProbeResult(RequestUtils.STATUS_CODE);
+        String contentType = requestResults.getProbeResult(RequestUtils.CONTENT_TYPE);
+
+        if (!(statusCode == HTTP_OK && contentType != null && OPENSEARCH_MIME_TYPES.contains(contentType))) {
```

Comment:
I looked into this, the null check was necessary due to a problem I was bubbling up.
I just handled the empty contentType header earlier in the process and was able to get rid of this check in all 3 Utils classes.